src/Session.php

```diff
@@ -0,0 +1,194 @@
+    }
+
+    /**
+      * Check if there is an active session
+      *
+      * 
+      *
+      * @return  boolean
+      *
+      */
+    public static function isActive()
+    {
+        return self::status() === PHP_SESSION_ACTIVE ? true : false;
+    }
+
+    /**
+     * Get session status
+     *
+     *
+     * @return constant
+     *
+     */
+    public static function status()
+    {
+        return session_status();
+    }
+
+    /**
+      * Start a new session if one is not already started
+      *
+      *
+      * @param boolean $supress_errors 
+      *
+      * @return  session_id || boolean(false)
+      *
+      */
+    public static function start($supress_errors=true)
+    {
+        if (!self::isActive()) {
+            if (!$session = $supress_errors ? @session_start() : session_start()) {
+                return false;
+            }
+        }
+
+        return self::getId();
+    }
+
+    /**
+      * Destroy the current session if one is active
+      *
+      * 
+      *
+      * @return  boolean
+      *
+      */
+    public static function end()
+    {
+        $destroyed = true;
+        if (self::isActive()) {
+            self::start();
+            $destroyed = session_destroy();
+        }
+
+        return $destroyed;
+    }
+
+    /**
+      * Replace the current session id with a new one
+      *
+      * @param boolean $delete Whether or not to delete the old session data
+      *
+      * @return  boolean
+      *
+      */
+    public static function refresh($delete=false)
+    {
+        return self::isActive() ? session_regenerate_id($delete) : false;
+    }
+
+    /**
+      * Set a single session variable
+      *
+      * @param string|int $key
+      * @param string|int|float|array|object|boolean $val
+      *
+      * @return  boolean(true)
+      *
+      */
+    public static function setVar($key, $val)
+    {
+        $_SESSION[$key] = $val;
+
+        return true;
+    }
+
+    /**
+      * Set multiple session variables
+      *
+      * @param array $vars
+      *
+      * @return  boolean(true)
+      *
+      */
+    public static function setVars(array $vars)
+    {
+        return array_replace_recursive($_SESSION, $vars) ? true : false;
+    }
+
+    /**
+      * Get a session variable if it exists
+      *
+      * @param string|int $key
+      *
+      * @return  $_SESSION[$key] || boolean(false)
+      *
+      */
+    public static function getVar($key, $default=null)
+    {
+        return self::isActive() && array_key_exists($key, $_SESSION) ? $_SESSION[$key] : (!is_null($default) ? $default : false);
+    }
+
+    /**
+      * Get all or a subset of session variables if session is active
+      *
+      * @param array $vars
+      *
+      * @return  array || boolean(false)
+      *
+      */
+    public static function getVars($vars=array())
+    {
+        return self::isActive() ? (!empty($vars) ? array_intersect_key($_SESSION, array_flip($vars)) : $_SESSION) : false;
```

Comment:
I agree with you initial comment about unpacking and skipping the active check. After reconsidering this function, it makes more sense to return all vars in the event that no vars are specified.